code/game/objects/structures/divine.dm

```diff
@@ -6,19 +23,47 @@
 	anchored = TRUE
 	density = FALSE
 	can_buckle = 1
+	max_integrity = 30000
+	var/divine_chance = 5
 
-/obj/structure/sacrificealtar/on_attack_hand(mob/living/user, act_intent = user.a_intent, unarmed_attack_flags)
+/obj/structure/sacrificealtar/attackby(obj/item/I, mob/user)
 	. = ..()
+	var/divine = prob(divine_chance)
 	if(.)
 		return
+	if(istype(I, /obj/item/gun))
+		to_chat(user, "<span class='notice'>You attempt to sacrifice [I] by invoking the sacrificial ritual, it melts and breaks into pieces.</span>")
+		qdel(I)
+		message_admins("[ADMIN_LOOKUPFLW(user)] has sacrificed [key_name_admin(I)] on the sacrificial altar at [AREACOORD(src)].")
+		switch(divine)
+			if(FALSE)
+				var/loot_item = pick(GLOB.gunparts_rates)
+				new loot_item (get_turf(user))
+				return
+			if(TRUE)
+				var/pick_adcomp = pickweight(GLOB.loot_craft_advanced)
+				new pick_adcomp(get_turf(user))
+				return
 	if(!has_buckled_mobs())
 		return
 	var/mob/living/L = locate() in buckled_mobs
 	if(!L)
 		return
-	to_chat(user, "<span class='notice'>You attempt to sacrifice [L] by invoking the sacrificial ritual.</span>")
-	L.gib()
-	message_admins("[ADMIN_LOOKUPFLW(user)] has sacrificed [key_name_admin(L)] on the sacrificial altar at [AREACOORD(src)].")
+	if(istype(I, /obj/item/clothing/accessory/talisman))
+		to_chat(user, "<span class='notice'>You attempt to sacrifice [L] by invoking the sacrificial ritual, during the sacrifice, you removed something from the stomach of the victim.</span>")
+		L.gib()
+		message_admins("[ADMIN_LOOKUPFLW(user)] has sacrificed [key_name_admin(L)] on the sacrificial altar at [AREACOORD(src)].")
+		switch(divine)
```

Comment:
I'd also like some kind of tier system to mobs, so a chicken isnt worth as much as a deathclaw. Plus, it could mean people spamming easy to make mobs for lots of chances at potions. Not really concerned about all the seeds though, maybe low-level mobs would just check for seeds?

Reply:
I don't see much point in this yet, since deathclaws and other large and dangerous mobs currently cannot be moved due to their size, which means there is no way to put them on the altar. The rest of the mobs are no different in their danger. To an equally mild degree, brahmins, chickens, or mirelurks, who have a habit of forming whole packs in front of the camp, can be worn on the altar. I will gladly try to add such a tier list when hunting for mobs becomes a real challenge, and not a routine task for 2 minutes

Reply:
Fair enough!